src/fileloader.cpp

```diff
@@ -11,25 +11,31 @@ namespace OTB {
 
 constexpr Identifier wildcard = {{'\0', '\0', '\0', '\0'}};
 
-Loader::Loader(const std::string& fileName, const Identifier& acceptedIdentifier) : fileContents(fileName)
+Loader::Loader(const std::string& fileName, const Identifier& acceptedIdentifier)
 {
+	try {
+		fileContents.open(fileName); // may throw if file does not exists
```

Comment:
I would rather pass the `std::ifstream` as a parameter so that it throws before the constructor is called, and in here it is guaranteed to be open.